common/lib/dependabot/dependency_file.rb

```diff
@@ -75,8 +86,12 @@ def support_file?
       @support_file
     end
 
+    def deleted
```

Comment:
This is not exactly the same as having an `attr_accessor`, the accessor also acts as a setter, we'll need to keep that behavior around unfortunately. Maybe we should reimplement that by setting the `@operation` to `DELETE`:

```ruby
def deleted=(deleted)
  operation = deleted ? Operation::DELETE : Operation::UPDATE
end
```

We should eventually deprecate the `deleted` field and rely on the `operation` instead, but we currently don't have a formal deprecation policy so I'll need to give that a think

Reply:
Ah, my bad. Not that much Ruby experience. 
That sounds like a good solution to me.

Can you tell me whether this one also need to be kept:
```
def deleted?
  deleted
end
```

Is that something different compared to `def deleted` now that it's not an `attr_accessor` anymore?

Reply:
Done :) And added a test for the setter.
Do we need to add any deprecation notices? I don't really see any through the codebase.

Reply:
Yeah we've historically never really deprecated many things, and when we did we've sort of pushed through breaking changes and dealt with them on our own side. I'll check-in with the team if we want to introduce a more thorough process